Okta.Core/Clients/AuthClient.cs

```diff
@@ -14,32 +15,93 @@ public class AuthClient : AuthenticatedClient
         public AuthClient(OktaSettings oktaSettings) : base(oktaSettings) { resourcePath = Constants.EndpointV1 + Constants.AuthnEndpoint; }
         public AuthClient(string apiToken, string subdomain) : base(apiToken, subdomain) { resourcePath = Constants.EndpointV1 + Constants.AuthnEndpoint; }
 
-        protected string resourcePath { get; set; }
-
+        protected string resourcePath { get; set; }
+
         /// <summary>
         /// Authenticates an Okta user
         /// </summary>
         /// <param name="username">User's username/login</param>
         /// <param name="password">User's password</param>
+        /// <param name="authContext">Object that alows trusted web applications to pass additional context for authentication</param>
         /// <param name="relayState">opaque value for the transaction and processed as untrusted data which is just echoed in a response. It is the client’s responsibility to escape/encode this value before displaying in a UI such as a HTML document </param>
         /// <param name="bWarnPasswordExpired">Optional parameter indicating whether the PASSWORD_WARN status should be returned if available. Defaults to false</param>
         /// <param name="bMultiOptionalFactorEnroll">Optional parameter indicating whether the user should be prompted to add an additional second factor if available </param>
-        /// <returns></returns>
-        public virtual AuthResponse Authenticate(string username, string password, string relayState = null, bool bWarnPasswordExpired = false, bool bMultiOptionalFactorEnroll = false)
-        {
-            var authRequest = new AuthRequest {
-                Username = username, 
-                Password = password, 
-                RelayState = relayState,
+        /// <returns></returns>
+        public virtual AuthResponse Authenticate(string username, string password, AuthContext authContext = null, string relayState = null, bool bWarnPasswordExpired = false, bool bMultiOptionalFactorEnroll = false)

```

Comment:
I think we should add the new `AuthContext` parameter at the end of the parameter list, so it's not a breaking change for existing calls.